tests/foreman/ui/test_ansible.py

```diff
@@ -336,3 +316,27 @@ def test_positive_remove_role_in_new_ui(self):
     :expectedresults: The Role is successfully removed from the Host, and no longer shows
         up on the UI
     """
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+
+    location = function_host.location.read()
+    organization = function_host.organization.read()
+    proxy_id = target_sat.nailgun_smart_proxy.id
+    target_sat.api.AnsibleRoles().sync(data={'proxy_id': proxy_id, 'role_names': [SELECTED_ROLE]})
+    role_id = target_sat.api.AnsibleRoles().search(query={'search': f'name={SELECTED_ROLE}'})[0].id
+    target_sat.api.Host(id=function_host.id).add_ansible_role(data={'ansible_role_id': role_id})
+    host_roles = function_host.list_ansible_roles()
+    assert host_roles[0]['name'] == SELECTED_ROLE
```

Comment:
Here, as this test is verifying both assign and remove, so I feel this part of role assign should also be part of below contextmanager to run via UI

Reply:
I agree, the removed stubbed test  `test_positive_assign_role_in_new_ui` indicated that the role would be assigned via UI, so that action should be taken in this test.

Reply:
One additional note, though: since Airgun does not provide a method for syncing a single role (only importing all of them at once), I think it's fine for syncing the role to remain an API call.